### PR TITLE
bitlbee: Add package

### PR DESCRIPTION
**Pull request by nikil**
Maintainer: me
Compile tested: ar71xx, x86_64
Run tested: x86_64

Description: An IRC to other chat networks gateway

Signed-off-by: Nikil Mehta nikil.mehta@gmail.com

### CODE REVIEW

File: net/bitlbee/Makefile [hunk 1]
```diff
@@ -0,0 +1,65 @@
+#
+# Copyright (C) 2006-2016 OpenWrt.org
```
Comment (diizzyy) [inline net/bitlbee/Makefile]: Please don't use OpenWrt.org copyright unless you are under contract.
https://www.lede-project.org/faq.html#who_holds_the_copyright_of_lede_code

File: net/bitlbee/Makefile [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/nls.mk
+
+define Package/bitlbee
+  SECTION:=net
+  CATEGORY:=Network
+  SUBMENU:=Instant Messaging
+  TITLE:=An IRC to other chat networks gateway
+  URL:=http://www.bitlbee.org/
+  DEPENDS:=+glib2 +libevent +libgcrypt +libgnutls $(ICONV_DEPENDS) $(INTL_DEPENDS)
```
Comment (diizzyy) [inline net/bitlbee/Makefile]: According to the FreeBSD ports tree you either use glib2 or libevent, not both.
http://www.freshports.org/irc/bitlbee/ (Configuration options)

File: net/bitlbee/Makefile [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=bitlbee
+PKG_VERSION:=3.4.2
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=http://get.bitlbee.org/src/
+PKG_MD5SUM:=3b620bf80c35d673cd08311792ec96be
```
Comment (diizzyy) [inline net/bitlbee/Makefile]: Use SHA256 instead of MD5

File: net/bitlbee/Makefile [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=bitlbee
+PKG_VERSION:=3.4.2
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=http://get.bitlbee.org/src/
```
Comment (diizzyy) [inline net/bitlbee/Makefile]: Apparently bitlbee's site can be a bit dodgy, here's a few mirrors:

http://distcache.FreeBSD.org/local-distfiles/brix/
http://distcache.eu.FreeBSD.org/local-distfiles/brix/
http://distcache.us-east.FreeBSD.org/local-distfiles/brix/
http://distcache.us-west.FreeBSD.org/local-distfiles/brix/

File: net/bitlbee/Makefile (hunk 2 above)
Comment (nikil) [inline net/bitlbee/Makefile]: Here's what the configure script reports when trying to use openssl; it looks like they really want bitlbee binaries to be distributed that link to gnutls.  Let me know what you think:

No detection code exists for OpenSSL. Make sure that you have a complete
installation of OpenSSL (including devel/header files) before reporting
compilation problems.

Also, keep in mind that the OpenSSL is, according to some people, not
completely GPL-compatible. Using GnuTLS is recommended and better supported
by us. However, on many BSD machines, OpenSSL can be considered part of the
operating system, which makes it GPL-compatible.

For more info, see: http://www.openssl.org/support/faq.html#LEGAL2
                    http://www.gnome.org/~markmc/openssl-and-the-gpl.html

Please note that distributing a BitlBee binary which links to OpenSSL is
probably illegal. If you want to create and distribute a binary BitlBee
package, you really should use GnuTLS instead.

Also, the OpenSSL license requires us to say this:
 *    "This product includes software developed by the OpenSSL Project
 *    for use in the OpenSSL Toolkit. (http://www.openssl.org/)"

**Comment by diizzyy** [comment]
I played around a little bit with this port and managed to fix a few things, libevent2 seems broken from what I can tell (doesn't compile, you need to use configure switch to enable it).

Full DEPENDS: list:
`DEPENDS:=+glib2 +libopenssl $(ICONV_DEPENDS) $(INTL_DEPENDS)`

Added configure args:
`        --msn=0 \
        --ssl=openssl
`

Best regards, 
Daniel

**Comment by nikil** [comment]
Those flags work great - thanks for your help!  I've updated the Makefile.

**Comment by diizzyy** [comment]
@hnyman @thess 
We already have irc bouncers in so I don't think this doesn't fit in the repo.

**Comment by thess** [comment]
@diizzyy - Please clarify the double-negative in your comment above. You are voting for in or out?

**Comment by diizzyy** [comment]
Merge it :)

**Comment by thess** [comment]
Thanks

**Pull request merged**